### ISSUE TITLE
Display chapter list based on student location so students can select a chapter

### ISSUE THREAD

**Issue by shaun-technovation** [Iridescent-CM/technovation-app#4408]
- Can only view/pick chapters with an active chapter affiliation agreement

**Comment by dboyer** [comment]
Depends on #4557 and #4821 

On the chapter selection screen, students should see a list of chapters near their location. We can use the existing system for identifying a chapter near a student as described in #4367.

**Comment by dboyer** [comment]
This involves:

- Checking student's location against chapter location, not ChA's location as happens in the current system
- Can continue to use the current method of how students are connected to chapters (look within region and then within country)

Questions to confirm with program team:
- Should we only display chapters that are fully onboarded, partially onboarded, or any chapter even if haven't onboarded at all
- Do we display all chapters for a country or just the chapter for a region if there is one for that student's region

**Comment by dboyer** [comment]
NOTE: Program Team would like any chapter with an active chapter affiliation agreement to be displayed in the list of chapters to select. The completion of other onboarding tasks does not impact display of the chapter.

**Comment by dboyer** [comment]
I'm guessing we will want to prompt them to review and confirm their chapter each season but that we wouldn't reset it, just like we don't reset a person's location. I'll confirm with PT.

**Comment by dboyer** [comment]
Tested on QA and confirmed:

- As a student, I registered in a city in Kenya with no chapters in the state. The list displayed both chapters in the country as options.
<img width="648" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd1pnQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--91df4e2e1e11fe888288d1a1b079926d834e8a87/Screen%20Shot%202024-10-21%20at%202.43.37%20PM.png" alt="Screen Shot 2024-10-21 at 2.43.37 PM.png" />

- As a student, I registered in a Kenya city with a chapter. Only that chapter appeared in the list of options.
<img width="1153" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd1ZnQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--19b3a96ef796cdae2e8f70226dfde7534bf9234f/Screen%20Shot%202024-10-21%20at%202.43.12%20PM.png" alt="Screen Shot 2024-10-21 at 2.43.12 PM.png" />

- As a student, I registered in a country with no chapters. I was given the "additional resources" pop-up.
